templates/Dockerfile-rpm.tmpl

```diff
@@ -93,6 +93,20 @@ RUN { \
     } \
     && yum clean all
 
+# install cmake, devtoolset and its dependencies to build azure sdk
+RUN yum -y install  perl-IPC-Cmd libuuid-devel cmake3
+
+# by default git 1.8.x is being installed in centos 7
+# Git 2.7.1 is minimum requirement for cmake so we remove and reinstall latests git from an up-to-date repo
+# Symbolic link is not being created for cmake from cmake3 by default. Therefore, we need to create the link as well.
+# devtoolset-8-gcc-c++ is required to compile azure sdk in pg azure storage project
+RUN [[ %%os%% != centos ]] || [[ %%release%% != 7 ]] || ( \
+    yum -y install  perl-IPC-Cmd libuuid-devel cmake3 && \
+    ln -s /usr/bin/cmake3 /usr/bin/cmake  && \
+    yum -y remove git && \
+    yum -y install https://packages.endpointdev.com/rhel/7/os/x86_64/endpoint-repo.x86_64.rpm && \
```

Comment:
I do not like how you installed a packaging repo here.

1. This does not follow official instructions from the owners.
2. There is not enough explanation on what endpoint repo is, and why it is necessary.
3. You do not verify the signing key, and this creates a vulnerability.
4. I'd be happier to see this repository removed after we install the minimal set of packages that are absent in official CentOS repositories.